### DISCUSSION TITLE
Add `--use-yarn` flag to `create-next-app`

### DISCUSSION THREAD

**Discussion by akd-io** [vercel/next.js#49405] [Ideas]
### Goals

1. Add `--use-yarn` flag to `create-next-app`, similar to `--use-npm` and `--use-pnpm`.

### Background

In [Create Next Stack](https://www.create-next-stack.com/), I currently have to fake the npm config user agent in order to force an install with Yarn.

Code in question is here: https://github.com/akd-io/create-next-stack/blob/develop/packages/create-next-stack/src/main/plugins/next.ts#LL84C29-L84C29

The reasoning for requesting support for `--use-yarn` is well described in the code. I quote:
> Below, we temporarily modify the `npm_config_user_agent` environment variable to make `create-next-app` use the correct package manager to install dependencies.
> This is done because `create-next-app` doesn't support `--use-yarn`.
> Instead, users of `create-next-app` are supposed to use `yarn create next-app` to use `create-next-app` with Yarn.
> This won't work for us though, as Yarn create doesn't support versioned package names, which we need to use to use the correct version of create-next-app.

That is, `yarn create next-app` isn't as powerful as `npx create-next-app`, so we need to call it using `npx`. But `create-next-app` does not support `--use-yarn`, so we hack it by calling `create-next-app` with a fake npm config user agent.

### Proposal

Steps:
- Implement `--use-yarn` flag
- Update docs
- Add tests

I am providing the following PR: https://github.com/vercel/next.js/pull/49407

**Comment by akd-io** [comment]
https://github.com/vercel/next.js/pull/49407 was merged

**Comment by konradbjk** [comment]
Guys it does not work, it does not install yarn properly. I am always getting yarn classic vs yarn berry
